what4/src/What4/Solver/Bitwuzla.hs

```diff
@@ -0,0 +1,103 @@
+-- Description      : Interface for running Bitwuzla
+-- Copyright        : (c) Galois, Inc 2014-2021
+-- License          : BSD3
+-- Maintainer       : Rob Dockins <rdockins@galois.com>
+-- Stability        : provisional
+--
+-- This module provides an interface for running Bitwuzla and parsing
+-- the results back.
+------------------------------------------------------------------------
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+module What4.Solver.Bitwuzla
+  ( Bitwuzla(..)
+  , bitwuzlaPath
+  , bitwuzlaOptions
+  , bitwuzlaAdapter
+  , bitwuzlaFeatures
+  , runBitwuzlaInOverride
+  ) where
+
+import           Data.Bits ( (.|.) )
+
+import qualified What4.BaseTypes as WT
+import qualified What4.Concrete as Concrete
+import qualified What4.Config as Config
+import           What4.Expr.Builder
+import           What4.Expr.GroundEval
+import qualified What4.Interface as WI
+import qualified What4.ProblemFeatures as Features
+import qualified What4.Protocol.Online as Online
+import qualified What4.Protocol.SMTLib2 as SMT2
+import qualified What4.SatResult as SatResult
+import qualified What4.Solver.Adapter as Adapter
+import qualified What4.Utils.Process as Process
+
+data Bitwuzla = Bitwuzla deriving Show
+
+-- | Path to bitwuzla
+bitwuzlaPath :: Config.ConfigOption (WT.BaseStringType WT.Unicode)
+bitwuzlaPath = Config.configOption WT.knownRepr "solver.bitwuzla.path"
+
+bitwuzlaOptions :: [Config.ConfigDesc]
+bitwuzlaOptions =
+  let bpopt co = Config.mkOpt co
+                              Config.executablePathOptSty
+                              (Just "Path to bitwuzla executable")
+                              (Just (Concrete.ConcreteString "bitwuzla"))
+      bp = bpopt bitwuzlaPath
+  in [bp] <> SMT2.smtlib2Options
+
+bitwuzlaAdapter :: Adapter.SolverAdapter st
+bitwuzlaAdapter = Adapter.SolverAdapter
+  { Adapter.solver_adapter_name = "bitwuzla"
+  , Adapter.solver_adapter_config_options = bitwuzlaOptions
+  , Adapter.solver_adapter_check_sat = runBitwuzlaInOverride
+  , Adapter.solver_adapter_write_smt2 =
+    SMT2.writeDefaultSMT2 () "Bitwuzla" Adapter.defaultWriteSMTLIB2Features
+    Nothing
+  }
+
+bitwuzlaFeatures :: Features.ProblemFeatures
+bitwuzlaFeatures = Features.useSymbolicArrays
+               .|. Features.useBitvectors
+               .|. Features.useFloatingPoint
+
+instance SMT2.SMTLib2Tweaks Bitwuzla where
+  smtlib2tweaks = Bitwuzla
+
+runBitwuzlaInOverride
+  :: ExprBuilder t st fs
+  -> Adapter.LogData
+  -> [BoolExpr t]
+  -> (SatResult.SatResult (GroundEvalFn t, Maybe (ExprRangeBindings t)) () -> IO a)
+  -> IO a
+runBitwuzlaInOverride =
+  SMT2.runSolverInOverride Bitwuzla SMT2.nullAcknowledgementAction
+     bitwuzlaFeatures Nothing
+
+instance SMT2.SMTLib2GenericSolver Bitwuzla where
+  defaultSolverPath _ = Process.findSolverPath bitwuzlaPath . WI.getConfiguration
+  defaultSolverArgs _ _ = return ["--smt2", "--model-gen", "--incremental", "--output-format=smt2"]
```

Comment:
I'm surprised that this doesn't just use the same arguments as in the Boolector configuration, given how similar it is to Bitwuzla. In particular, the absence of `-e=0` seems off, since I'm fairly certain things won't work as intended if `unsat` results cause non-zero exit codes.